fuse/eval/evaluator.py

```diff
@@ -211,13 +214,29 @@ def read_data(
                 df_list.append(data_elem_df)
 
             # make sure ids exists in all dataframes
-            for data_elem_df in df_list:
+            for i_df, data_elem_df in enumerate(df_list):
```

Comment:
This case is for an ensemble where all input dataframe are computed on the same dataset (or largely overlapping datasets). In a sequence of folds the folds are disjoint - so I don't see the connection to this code. Anyway, I don't want to add changes in parts that my code does not use, to avoid adding bugs.